File: CS/WindowsApplication1/MyRibbonControl/MyRibbonPageGroupComplexLayoutCalculator.cs
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace WindowsApplication1
{
    public class MyRibbonPageGroupComplexLayoutCalculator : RibbonPageGroupComplexLayoutCalculator
    {

        public MyRibbonPageGroupComplexLayoutCalculator(RibbonPageGroupViewInfo groupInfo)
            : base(groupInfo)
        {

        }

        public MyRibbonViewInfo ViewInfo
        {
            get
            {
                return GroupInfo.ViewInfo as MyRibbonViewInfo;
            }
        }

        protected override int Reduce()
        {
            if ( GroupInfo.Items.Count == 0 )
                return 0;

            if ( !ViewInfo.CanMinimizePageGroup(GroupInfo.PageGroup) )
                return 1;

            return base.Reduce();
        }

    }
}

File: VB/WindowsApplication1/MyRibbonControl/MyRibbonPageGroupComplexLayoutCalculator.vb
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace WindowsApplication1
	Public Class MyRibbonPageGroupComplexLayoutCalculator
		Inherits RibbonPageGroupComplexLayoutCalculator

		Public Sub New(ByVal groupInfo As RibbonPageGroupViewInfo)
			MyBase.New(groupInfo)
		End Sub

		Public ReadOnly Property ViewInfo() As MyRibbonViewInfo
			Get
				Return TryCast(GroupInfo.ViewInfo, MyRibbonViewInfo)
			End Get
		End Property

		Protected Overrides Function Reduce() As Integer
			If GroupInfo.Items.Count = 0 Then
				Return 0
			End If

			Dim pageGroupInfo As RibbonPageGroupViewInfo = TryCast(GroupInfo, RibbonPageGroupViewInfo)
			If pageGroupInfo Is Nothing Then
				Return MyBase.Reduce()
			End If

            If (Not ViewInfo.CanMinimizePageGroup(pageGroupInfo.PageGroup)) Then
                Return 1
            End If

			Return MyBase.Reduce()
		End Function
	End Class
End Namespace
